vindi-woocommerce-subscriptions.php

```diff
@@ -226,43 +264,39 @@ public function save_subscription_meta($post_id)
             }
 
             update_post_meta($post_id, 'vindi_subscription_plan', $subscription_plan);
-
-
-            if(preg_match('/variable-subscription/', $wc_product->get_type())) {
-                foreach ($wc_product->get_children() as $child) {
-                    update_post_meta($child, '_subscription_length', $subscription_length);
-                    if($subscription_period_interval % 12 == 0) {
-                        update_post_meta($child, '_subscription_period_interval', $years_interval);
-                        update_post_meta($child, '_subscription_period', 'year');
-                    } else {
-                        update_post_meta($child, '_subscription_period_interval', $subscription_period_interval);
-                        update_post_meta($child, '_subscription_period', $subscription_period);
-                    }
-                }
-            }
         }
 
-        public function save_ajax_subscription_meta($post_id)
+        /**
+         * @param int $variation_id
+         */
+        public function save_subscription_variation_meta($variation_id, $i)
         {
-            if (false === $this->is_product_type_from_post(['variable-subscription'])) {
-                return;
+            $child_subscription_plan            = wc_clean($_POST['vindi_variable_subscription_plan'][$i]);
+            $child_subscription_period_interval = wc_clean($_POST['variable_subscription_period_interval'][$i]);
+            $child_subscription_period          = wc_clean($_POST['variable_subscription_period'][$i]);
+            $child_subscription_length          = wc_clean($_POST['variable_subscription_length'][$i]);
+
+            update_post_meta($variation_id, '_subscription_length', $child_subscription_length);
+            if($child_subscription_period_interval % 12 == 0) {
```

Comment:
```suggestion
            if ($child_subscription_period_interval % 12 == 0) {
```

Podemos escrever assim?

Reply:
Sim